pkg/plugin/pipedsdk/deployment.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipedsdk
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	config "github.com/pipe-cd/pipecd/pkg/configv1"
+	"github.com/pipe-cd/pipecd/pkg/plugin/api/v1alpha1/deployment"
+	"github.com/pipe-cd/pipecd/pkg/plugin/logpersister"
+	"github.com/pipe-cd/pipecd/pkg/plugin/pipedapi"
+)
+
+var (
+	deploymentServiceServer interface {
+		Plugin
+
+		Register(server *grpc.Server)
+		setCommonFields(commonFields)
+		deployment.DeploymentServiceServer
+	}
+)
+
+// DeploymentPlugin is the interface that be implemented by a full-spec deployment plugin.
+// This kind of plugin should implement all methods to manage resources and execute stages.
+type DeploymentPlugin[Config any] interface {
```

Comment:
The `Config` parameter is the plugin's config defined in piped's config.

Reply:
I added config on this commit.
https://github.com/pipe-cd/pipecd/pull/5529/commits/337e8c864d492a95c0d87608400e1a0fc93898c8